### PR TITLE
Generate self signed ssl cert and enable git auth

### PR DESCRIPTION
**Pull request by mdsufair**
- Disabled access through the public domain and removed DNS & route53 zones
- Generate a self-signed SSL certificate and import it to AWS ACM
- Option to enable auth config while dkubex installation which supports github/okta/google
- Option to enable ldap server which auto adds users
- Added readme file

### CODE REVIEW

File: dkubex/terraform/install.tf [hunk 1]
```diff
@@ -52,27 +62,32 @@ resource "null_resource" dkubex {
...
                 --set nfs.nfs_server="$NFS:/" \
                 --set nfs.storage_path="/" \
                 --set registry.username="dkubex123" \
                 --set registry.password=$REGISTRY_PASSWORD \
                 --set storage_type="nfs" \
                 --set wipe_data=false \
-                --set loadbalancer.eks.internal_lb=$INTERNAL_LB
+                --set loadbalancer.eks.internal_lb=$INTERNAL_LB \
+                --set auth.client_id=$AUTH_CLIENT_ID \
+                --set auth.client_secret=$AUTH_CLIENT_SECRET \
```
Comment (songole) [inline dkubex/terraform/install.tf:75]: Can we not hardcode this to github based auth? Can we not do it from the UI?

Comment (mdsufair) [reply dkubex/terraform/install.tf:75]: Made it a variable so that users can configure it using github/okta/others

File: dkubex.tf [hunk 2]
```diff
@@ -26,6 +26,7 @@ module "dkubex" {
   source                  = "./dkubex/terraform"
   registry_password       = var.registry_password
   auth_provider           = "github"
+  auth_issuer_url         = ""
```
Comment (songole) [inline dkubex.tf:29]: if I want to enable auth later, is it supported? Can I specify None to auth_provider ?

**Comment by songole** [comment]
Can you create a README with info about how to obtain some of the required config for gmail/github authentication

File: README.md [hunk 3]
```diff
@@ -1 +1,50 @@
...
+Initialise terraform by `terraform init` which will download all the providers and modules required for the scripts.
+
+
+### Setup variables
+1. You need to provide registry token which is required by the scripts to download images, by:
+
+        export TF_VAR_registry_password="$registry_token_provided_to_you"
+2. Set the release version of DkubeX to be installed by editing `release` variable in `dkubex.tf` file.
+3. You can enable auth config if you have a precreated OAuth App. We support OAuth App by github, okta and google as of now. If not you need to manually setup auth config after the installation.\
+To enable auth config as a part of installation edit `auth_enabled` variable to `true`, along with `auth_provider` variable to `github/okta/google`, `auth_issuer_url` variable and `auth_email_domain` variable as per your OAuth App in `dkubex.tf` file.\
```
Comment (songole) [inline README.md:23]: Can you add some info about how to obtain the required config for github and google providers?

**Pull request merged**